### ISSUE TITLE
Need another release~

### ISSUE THREAD

**Issue by sleagon** [tealeg/xlsx#456]
The last release 1.0.3 was released in 2017, is there any plan to release a new version, like 1.0.4 or 1.1.0?  There is a lot of commits merged to master since 2017.

**Comment by plandem** [comment]
I know, it's not good practice to advertise another library, but here is another one for a quite long time already: https://github.com/plandem/xlsx

And yes, Im author of that lib.

**Comment by ShawnMilo** [comment]
@plandem: What's the difference between your library and this one? Is it originally based off of this, a fresh rewrite, or something else?

I don't think it's bad practice to mention alternatives. It would be more helpful for users if they understood why one might choose one over another. Ultimately, the way Go package imports work, there's not a huge advantage for one library having a more recent official release, as `go get` is very simple, as is vendoring.

**Comment by plandem** [comment]
it's not based or used any part of code. It's totally written from ground zero.

I started project, because I needed to update xlsx file that had a lot formatting created by "managers", so file should be loaded - get some data, load to database info for BI, update some values in xlsx and sent back to managers without losing any formatting. That was the main reason - keep formatting as is.

**Comment by plandem** [comment]
and right now, tbh, library that I mentioned has more features

**Comment by sleagon** [comment]
@plandem we choose this library because of the stars, we thought it is more reliable. The feature we need most is stream R/W, every time parsing huge excel file, our server suffers. It seems like your project provides the ability of stream reading, I'll try your project and thanks for your kind advise~~

**Comment by plandem** [comment]
@sleagon sadly, that a true story - easier to get stars as first solution :)

P.S.: while read streaming is working, I would not expect a lot of optimization due to fact that zip files `in memory`.  At least I could not get a huge difference.  Although overall memory usage should be lower  anyway, due to architecture of library. 

P.P.S.: There is a task in roadmap to unpack on disk and stream only required files from disk.

**Comment by ryho** [comment]
In some sense, I am the current maintainer of this library. I have write permission and review PRs and make updates as needed, anyway.

I’ve never used the release feature, I don’t really feel a need for it. We use the master branch of the library, and I don’t merge PRs with out running all of our unit tests for our use of its Excel import/export features. That has not been an issue for us yet, and our users would immediately be very vocal if there was a problem.

We use this library’s stream write feature that I added, and it has worked well for us. This library also has the feature of “only read the first X rows of a file”, which we use for quick previews. No ability to stream read the full file, though that could be added. We also see problems when reading huge files, the memory usage of the Go XML library is insane. However, we are resilient to a host going down and files with 1M rows (Excel max) almost always go through, so it’s not a pain point.

PS: the community should always encourage frank discussion of alternatives, and of a library’s strengths and weaknesses.

**Comment by sleagon** [comment]
@ryho I do not want to argue about whether always using master is good choice or not, even you do not need a new version, most of other users may need that. If you are using go mod, you will never get the latest version of this package unless manually import the master. After all, you are the maintainer, it is up to you~ Feel free to close this issue.

**Comment by ryho** [comment]
>After all, you are the maintainer, it is up to you

I don’t like this way of thinking.
I do know that my maintenance is insufficient for this project, but I cannot justify to myself and to my company any more than the time I spend it on it now.

I have done a release for you. The version I’ve released is used at scale at Square, and I’d fix a bug in its basic data input and output in less than 24hr notice. I do not care about any issue in text formatting or font, however.

I’m open to additional maintainers to this project, if you are also open to that role. I did not start this project. I contributed enough high quality features that I was granted write access. I’m just the only person with write access that still cares. I didn’t ask for that.

**Comment by sleagon** [comment]
@ryho I mean no harm, thanks very much for your contribution.

**Comment by ryho** [comment]
In addition, I have thoughts about using a library’s master.
When I make a code change, it gets reviewed. My coworkers look at it. All unit tests must run. We’ve got people that test the change.  Even still, it lives on the staging data center (I’m privileged, we also have so many company internal users that we’d obviously hear about an issue after a few hours of a bad sha being on the staging DC, ) and then we wait and deploy to prod. We only deploy two of the nodes at first, to be sure the rest can still take the traffic. If at any step there is an issue, we roll back. We are in a unique position where theoretically the staging data center has no sla, but I get fucked if out staging nodes are down for 30 min, because so many people are testing things that depend on our latest changes

**Comment by tealeg** [comment]
Just to add to what @ryho has said.  I started this project many, many moons ago when Go was young.  I wrote it for my own purposes, which it fulfilled very well, until late 2011.  In the 8 years since then I have only once had need of it for something I was working on, and despite that, I committed a fair amount of time to maintain it.  Around the time my 2nd daughter was born I simply couldn't keep up with it anymore, and I gave both @ShawnMilo and @ryho commit rights.   For what it's worth, a great many people have contributed, and @ryho 's streaming work is a massive and useful contribution.   Any open source project is only as good as the people committing time to it.   If there are better XLSX libraries out there, I don't think anyone has an issue with people saying so (I even linked to an alternative right at the top of this repo).  .. in the meantime please don't get on anyone's back about this library, everyone giving time here is providing that service to you free of charge ;-)

**Comment by sleagon** [comment]
@ryho @tealeg Maybe you guys misunderstood me, i just want to make a kind advise for another release, nothing more. I used go mod to add this package to my project at first, however i found there is no Method exported related to streaming APIs. Then I found there is no version newer than 1.0.3 which was published at 2017. This package is great and the stream APIs contributed by ryho is wonderful, I just want them be tried by more people. My English is not very good, so i prefer to use some simple words, which may make you think i am complaining, sorry for that.
